### ISSUE TITLE
KasmVNC on Fedora 37, how to install ?

### ISSUE THREAD

**Issue by luc-languagetools** [kasmtech/KasmVNC#127]
KasmVNC works flawlessly for me on Ubuntu 22.04, but I'd like to install it on Fedora. Is the CentOS rpm expected to work ? It reported missing dependencies for me and wouldn't install. Has anyone successfully installed on Fedora 37 ? I'm now looking into building from source, but it seems to be a very arduous process.

**Comment by ben-alkov** [comment]
https://github.com/kasmtech/KasmVNC/releases/download/v1.1.0/kasmvncserver_fedora_thirtyseven_1.1.0_x86_64.rpm is installable on F38, assuming you don't mind deleting 'anaconda' and 'anaconda-gui' (things which pull in 'tigervnc-server-minimal')

```bash
🐚 sudo dnf install ./kasmvncserver_fedora_thirtyseven_1.1.0_x86_64.rpm
[sudo] password:
Error:
 Problem: problem with installed package tigervnc-server-minimal-1.13.1-3.fc38.x86_64
  - package kasmvncserver-1.1.0-1.fc37.x86_64 from @commandline conflicts with tigervnc-server-minimal provided by tigervnc-server-minimal-1.13.1-3.fc38.x86_64 from @System
  - package kasmvncserver-1.1.0-1.fc37.x86_64 from @commandline conflicts with tigervnc-server-minimal provided by tigervnc-server-minimal-1.13.1-3.fc38.x86_64 from fedora
  - conflicting requests
(try to add '--allowerasing' to command line to replace conflicting packages or '--skip-broken' to skip uninstallable packages)
```

At a minimum, there's a conflict in that both 'tigervnc-server-minimal' and 'kasmvncserver' provide `vncserver`

The complete install command: `🐚 sudo dnf install ./kasmvncserver_fedora_thirtyseven_1.1.0_x86_64.rpm  --allowerasing`

OTOH, trtying to set up `vncsever` fails when trying to find `select-de.sh`. I'm digging around in the Perl now (ew) - I might report more later...

\*sigh\* it's later...

`vncsever` is looking for 'bin/../builder/startup/deb/select-de.sh', which actually exists in '/usr/lib/kasmvncserver'

tl;dr `IsThisSystemBinary` below should be true on Fedora - don't know why it's not

```perl
sub DetectSelectDeBin {
  if (IsThisSystemBinary()) {
    "/usr/lib/kasmvncserver/select-de.sh";
  } else {
    LocalSelectDePath();
  }
}

sub LocalSelectDePath {
  my $dirname = dirname($0);
  "$dirname/../builder/startup/deb/select-de.sh";
}
```

**Comment by ledestin** [comment]
@ben-alkov what does `type vncserver` say?

**Comment by ben-alkov** [comment]
```bash
🐚 type vncserver
vncserver is /bin/vncserver
```

**Comment by ledestin** [comment]
@ben-alkov it's a bug, a workaround would be to call it as `/usr/bin/vncserver`. I think on modern systems both `/bin` and `/usr/bin` point to the same place.

**Comment by ben-alkov** [comment]
> I think on modern systems both `/bin` and `/usr/bin` point to the same place.

For all systems with which I'm familiar, yes

**Comment by antofthy** [comment]
I also installed   kasmvncserver_fedora_thirtyeight_1.2.0_x86_64.rpm
when I try to run  `vncserver` I get
```
Can't exec "/bin/../builder/startup/deb/select-de.sh": No such file or directory at /bin/vncserver line 827.
Failed to execute /bin/../builder/startup/deb/select-de.sh
```
The path should be `/usr/lib/kasmvncserver/select-de.sh`

The reason is simple....   in `/bin/vncserver`  we have
```perl
sub IsThisSystemBinary {
  $0 =~ m!^/usr!;
}
```
except the path to this binary is  `/bin/vncserver`  and NOT `/usr/bin/vncserver`
That is because quite a few releases of fedora ago the paths `/bin` and `/usr/bin` were merged.
```bash
$ ls -Fla /bin
lrwxrwxrwx 1 root root 7 Jan 19  2023 /bin -> usr/bin/
```
However a LOT of user PATH variables use `/bin` before `/usr/bin`
As such `$0` is set to `/bin/vncserver`

Script needs to understand BOTH paths if it wants to determine if script is a system installed version.
```perl
sub IsThisSystemBinary {
  $0 =~ m!^(/usr/bin/|/bin/)!;
}
```
